specification.markdown

```diff
@@ -1,66 +1,140 @@
 
--   Signed Integer - Big-endian, any size from 1 to 8 octets
--   Unsigned Integer - Big-endian, any size from 1 to 8 octets
--   Float - Big-endian, defined for 4 and 8 octets (32, 64 bits)
--   String - Printable ASCII (0x20 to 0x7E), zero-padded when needed
--   UTF-8 - [Unicode string](http://www.unicode.org/), zero padded when
-    needed ([RFC 2279](http://www.faqs.org/rfcs/rfc2279.html))
--   Date - signed 8 octets integer in nanoseconds with 0 indicating the
-    precise beginning of the millennium (at
-    2001-01-01T00:00:00,000000000 UTC)
--   master-element - contains other EBML sub-elements of the next lower
-    level
--   Binary - not interpreted by the parser
+Data Type        | Definition
+-----------------|----------------------------------------------------
+Signed Integer   | Big-endian, any size from 1 to 8 octets
+Unsigned Integer | Big-endian, any size from 1 to 8 octets
+Float            | Big-endian, defined for 4 and 8 octets (32, 64
+                 | bits)
+String           | Printable ASCII (0x20 to 0x7E), zero-padded when
+                 | needed
+UTF-8            | [Unicode string](http://www.unicode.org/), zero
+                 | padded when needed ([RFC 2279](http://www.faqs.org/rfcs/rfc2279.html))
+Date             | signed 8 octets integer in nanoseconds with 0
+                 | indicating the precise beginning of the
+                 | millennium (at 2001-01-01T00:00:00.000000000 UTC)
+Master-element   | contains other EBML sub-elements of the next lower
+                 | level
+Binary           | not interpreted by the parser
 
 As well as defining standard data types, EBML uses a system of Elements
-to make up an EBML "document." Elements incorporate an Element ID, a
-descriptor for the size of the element, and the binary data itself.
-Futher, Elements can be nested, or contain, Elements of a lower "level."
+to make up an EBML "document". Elements incorporate an Element ID, a
+descriptor for the size of the element (known as the Element Data
+Size), and the binary data itself. Futher, Elements can be nested or
+contain Elements of a lower "level".
+
 
-Element IDs are outlined as follows, beginning with the ID itself,
-followed by the Data Size, and then the non-interpreted Binary itself:
+### Element ID
+
+The EBML Element ID is encoded as a variable size integer with, by
+default, widths up to 4. Another maximum width value can be set by
+setting another value to EBMLMaxIDLength in the EBML header. See
+section 5.1. Element IDs are always encoded in their shortest form, e.g. 1 is always encoded as 0x81 and never as 0x4001. This limits the number of Element IDs in every class with the number of Element IDs in the previous classes. Furthermore, values with all data bits set to 1 and all data bits set to 0 are reserved, hence the effective number of Element IDs are as follows for different widths. Note that the shortest encoding form for 127 is 0x407f since 0x7f is reserved.
```

Comment:
It's coming from the RFC; neither the old EBML specs nor the Matroska specs contain such a limiation. So yes, we should move that to a separate pull request, please, as backwards compatibility should be considered.